baseplate/experiments/__init__.py

```diff
@@ -171,66 +183,19 @@ def variant(self, name, user=None, bucketing_event_override=None,
 
         if do_log:
             assert bucketing_id
-            self._log_bucketing_event(experiment, variant, user,
-                                      extra_event_fields)
+            if self._event_logger:
+                self._event_logger.log(
+                    experiment=experiment,
+                    variant=variant,
+                    user_id=inputs.get('user_id'),
+                    logged_in=inputs.get('logged_in'),
+                    cookie_created_timestamp=inputs.get('cookie_created_timestamp'),
+                    app_name='r2'
```

Comment:
We shouldn't have this value hardcoded in baseplate, we should pass this value to the `ContextFactory` which will pass it to the `Experiments` client.

Reply:
agreed, but I'm going to disagree about putting in the ContextFactory. I think it makes sense to pass into the variant method, since it can be dynamic even in the context of a single Experiment instance.

I'm thinking if relevance, for instance, was trying out a new algo, they could pass along the app_name a user is accessing from for each variant call.

Make sense?

Reply:
I guess I'm not even sure what `app_name` even means?  Would it really be that dynamic?  Where would the developer even get it from?

Reply:
the app_name is meant to pass in the context around what service/app is checking experimental variants. Might be a better name for it though.

The more I think about it, the more I think it doesn't belong here at all, and should be the responsibility of the EventLogger to populate, and/or as an optional field pulled from the extra_event_fields.